### PR TITLE
Fix for return code checking on ConfirmSignature

### PR DESCRIPTION
**Pull request by dgarske**
Now it returns actual error codes or 0 on success to capture all error cases.

### CODE REVIEW

**Comment by kaleb-himes** [comment]
retest this please jenkins, windows slave was offline when commit made.

**Comment by kaleb-himes** [comment]
Requesting review of test-case from @toddouska

**Comment by dgarske** [comment]
Idea: Why not just use a DER format SSL_FILETYPE_ASN1 and make a copy of the certs_test.h buffer and modify the last byte to corrupt the signature? Then you can remove these bad certs in the filesystem.

**Comment by kaleb-himes** [comment]
Great idea! Thanks David.

**Comment by dgarske** [comment]
@toddouska : This one is ready for final review. Its the same original fix, but I've cleaned up the tests a bit. I choose to keep the fix with the test case. Thanks.

File: tests/api.c [hunk 1]
```diff
@@ -3129,6 +3135,111 @@ static void test_wc_ecc_get_curve_id_from_name(void)
...
+
+        ret = wolfSSL_CertManagerVerifyBuffer(cm, cert_buf, cert_sz, SSL_FILETYPE_ASN1);
+        /* Let AssertIntEQ handle return code */
+
+        wolfSSL_CertManagerFree(cm);
+
+        return ret;
+    }
+
+    static int test_RsaConfirmSig_cm(void)
```
Comment (toddouska) [inline tests/api.c]: Is this naming right?  test_RsaConfirmSig_cm suggests a signature confirmation but it's trying to detect signature failures.  Shouldn't we have both?

File: tests/api.c [hunk 2]
```diff
@@ -3129,6 +3135,111 @@ static void test_wc_ecc_get_curve_id_from_name(void)
...
+
+        printf("Verify signature exception test: RSA: Ret %d\n", ret);
+
+        if (cert_buf)
+            free(cert_buf);
+
+        return ret;
+    }
+
+    static int test_EccConfirmSig_cm(void)
```
Comment (toddouska) [inline tests/api.c]: same as RSA naming.

**Pull request merged**